### PR TITLE
feat(Bonus Pagamenti Digitali): [#175269179] Transaction details

### PR DESCRIPTION
**Pull request by fabriziofff**
## Short description
This pr adds the bottomsheet with details for a transaction.

Normal            |  MaxTransactionCashback
:-------------------------:|:-------------------------:
|  

MaxPeriodCashback            |  CancelTransaction
:-------------------------:|:-------------------------:
|  

## List of changes proposed in this pull request
- Added `keyId: string;  maxCashbackForTransactionAmount: number | undefined;` to `EnhancedBpdTransaction`.
- Added `BpdTransactionDetailComponent`  to render a transaction detail.
- Added 

## How to test
Describe the tests that you ran to verify your changes. Provide instructions so we can reproduce.

### CODE REVIEW

File: ts/features/bonus/bpd/screens/details/transaction/detail/BpdTransactionDetailComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+      </H5>
+      <View style={styles.rowId}>
+        {props.transaction.amount < 0 && (
+          <>
+            <CancelBadge />
+            <View hspacer={true} />
+          </>
+        )}
+        <H4 weight={"SemiBold"} color={"bluegreyDark"}>
+          {`€ ${formatNumberAmount(props.transaction.amount)}`}
```
Comment (Undermaken) [inline ts/features/bonus/bpd/screens/details/transaction/detail/BpdTransactionDetailComponent.tsx:89]:
```suggestion
          {formatNumberAmount(props.transaction.amount, true)}
```

File: ts/features/bonus/bpd/screens/details/transaction/detail/BpdTransactionDetailComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        </H4>
+      </View>
+    </View>
+    <View spacer={true} small={true} />
+    <View style={styles.rowId}>
+      <H5 weight={"Regular"} color={"bluegrey"}>
+        {I18n.t("bonus.bpd.details.transaction.detail.cashbackAmount")}
+      </H5>
+      <H4 weight={"SemiBold"} color={"bluegreyDark"}>
+        {`€ ${formatNumberAmount(props.transaction.cashback)}`}
```
Comment (Undermaken) [inline ts/features/bonus/bpd/screens/details/transaction/detail/BpdTransactionDetailComponent.tsx:99]:
```suggestion
        {formatNumberAmount(props.transaction.cashback, true)}
```

File: ts/features/bonus/bpd/screens/details/transaction/detail/BpdTransactionWarning.tsx [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+          <Body>
+            {I18n.t("bonus.bpd.details.transaction.detail.maxCashbackWarning", {
+              amount: props.transaction.maxCashbackForTransactionAmount
+            })}
+          </Body>
+        </InfoBox>
+      </>
+    );
+  }
+  if (props.transaction.cashback === 0) {
```
Comment (Undermaken) [inline ts/features/bonus/bpd/screens/details/transaction/detail/BpdTransactionWarning.tsx:33]:
```suggestion
if (props.transaction.cashback <= 0) {
    return (
      <>
        <View spacer={true} />
        <InfoBox>
          <Body>
            {I18n.t(
              props.transaction.cashback < 0
                ? "bonus.bpd.details.transaction.detail.canceledOperationWarning"
                : "bonus.bpd.details.transaction.detail.maxCashbackForPeriodWarning"
            )}
          </Body>
        </InfoBox>
      </>
    );
  }
```
Could it be a valid refactoring?

**Review by Undermaken — CHANGES_REQUESTED** [review]

File: ts/features/bonus/bpd/screens/details/transaction/detail/BpdTransactionWarning.tsx (hunk 3 above)
Comment (fabriziofff) [reply ts/features/bonus/bpd/screens/details/transaction/detail/BpdTransactionWarning.tsx]: yes, better, changed in f801036

**Review by Undermaken — APPROVED** [review]

**Pull request merged**